### PR TITLE
File popup mariana

### PR DESCRIPTION
**Pull request by mariana-novoa**
Hi so in this branch we worked on creating a right-click menu for files and directories. The popup menu includes create Text File, delete File, copy and paste. It shows you a different one whether you select a directory or a file. Additionally, since Dario also had a right click popup menu for the management of files (close all files, etc), we decided to incorporate both into a single class to adhere to Clean Architecture. Note that we are missing some Javadoc comments.

### CODE REVIEW

**Review by dariosanp — APPROVED** [review]
The new code seems to merge with no issues. The conflicts seem to be caused by removing unnecessary code. I Approve of adding your version.

File: src/main/DarioMarianaTest/use_case/EditorOperations/EditorOperationsTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        assertTrue(EditorOperations.isDuplicate(removeFile, tabbedPane));
+
+        // Clean up the temporary files
+        for (File tempFile : tempFiles) {
+            if (tempFile.exists()) {
+                tempFile.delete();
+            }
+        }
+    }
+
```
Comment (marburcea) [inline src/main/DarioMarianaTest/use_case/EditorOperations/EditorOperationsTest.java:79]: Great test case and coverage, but make sure to move it to test file before merging!

File: src/main/java/entity/ParentIDEJTabbedPane.java [hunk 2]
```diff
@@ -38,94 +35,17 @@ public void mousePressed(MouseEvent e) {
...
-        }
-    }
-
-    public void closeTabsToLeft(int index) {
-        while (index > 0) {
-            closeTab(0);
-            index--;
-        }
-    }
-
```
Comment (marburcea) [inline src/main/java/entity/ParentIDEJTabbedPane.java:125]: Way to move the logic for this code to another file and make it more abstract

File: src/main/java/view/PopupMenuHandler.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                JOptionPane.showMessageDialog(null, "Failed to delete: " + ex.getMessage(), "Delete Error", JOptionPane.ERROR_MESSAGE);
+            }
+        });
+
+        popupMenu.add(newFileItem);
+        popupMenu.add(copyDirItem);
+        popupMenu.add(pasteDirItem);
+        popupMenu.add(renameDirItem);
+        popupMenu.add(deleteDirItem);
+
```
Comment (marburcea) [inline src/main/java/view/PopupMenuHandler.java:168]: improved way to handle a pop up

**Comment by marburcea** [comment]
Approved overall

**Comment by yaraalkurdi** [comment]
Approved

File: src/main/java/entity/Editor.java [hunk 4]
```diff
@@ -1,34 +0,0 @@
-package entity;
```
Comment (AriGood) [inline src/main/java/entity/Editor.java:1]: by removing this are we just pushing logic downwards

**Review by AriGood — CHANGES_REQUESTED** [review]
.idea folder should not be here please don't merge well it is there

also you should add docstrings and error handling before merging

File: src/main/java/entity/Editor.java (hunk 4 above)
Comment (dariosanp) [reply src/main/java/entity/Editor.java:1]: Editor served no purpose as it was replaced by EditorObj

**Pull request merged**